lib/fluent/plugin/in_tail.rb

```diff
@@ -548,7 +636,7 @@ def detach_watcher_after_rotate_wait(tw, ino)
       if @open_on_every_update
         # Detach now because it's already closed, waiting it doesn't make sense.
         detach_watcher(tw, ino)
-      elsif @read_bytes_limit_per_second < 0
+      elsif @read_bytes_limit_per_second < 0 || (!tw.group_watcher.nil? && tw.group_watcher.limit <= 0)
```

Comment:
Could you extract this condition as a method for readability and testability?

Reply:
I have changed it to: 
```ruby
elsif @read_bytes_limit_per_second < 0 || tw.group_watcher&.limit <= 0
```